write-fonts/src/tables/cmap.rs

```diff
@@ -2,62 +2,142 @@
 //!
 //! [cmap]: https://docs.microsoft.com/en-us/typography/opentype/spec/cmap
 
+use std::collections::HashSet;
+
 include!("../../generated/generated_cmap.rs");
 
+// https://learn.microsoft.com/en-us/typography/opentype/spec/cmap#unicode-platform-platform-id--0
+const UNICODE_BMP_ENCODING: u16 = 3;
+
+fn size_of_cmap4(seg_count: u16, gid_count: u16) -> u16 {
+    8 * 2  // 8 uint16's
+    + 2 * seg_count * 4  // 4 parallel arrays of len seg_count, 2 bytes per entry
+    + 2 * gid_count // 2 bytes per gid in glyphIdArray
+}
+
+impl CmapSubtable {
+    fn create_format_4(
+        lang: u16,
+        end_code: Vec<u16>,
+        start_code: Vec<u16>,
+        id_deltas: Vec<i16>,
+    ) -> Self {
+        if HashSet::from([start_code.len(), end_code.len(), id_deltas.len()]).len() != 1 {
+            panic!("uneven parallel arrays, very bad. Very very bad.");
+        }
```

Comment:
I would probably just write this as,

```rust
assert!(end_code.len() == start_code.len() && end_code.len() == id_deltas.len(), "msg");
```

this doesn't really matter, but importing HashSet and doing an allocation here just seems unnecessary.